bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHAApplianceHandler.java

```diff
@@ -0,0 +1,426 @@
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tec.uom.se.unit.Units;
+
+/**
+ * The {@link PlugwiseHAApplianceHandler} class is responsible for handling
+ * commands and status updates for the Plugwise Home Automation appliances.
+ * Extends @{link PlugwiseHABaseHandler}
+ *
+ * @author Bas van Wetten - Initial contribution
+ * @author Leo Siepel - finish initial contribution
+ *
+ */
+@NonNullByDefault
+public class PlugwiseHAApplianceHandler extends PlugwiseHABaseHandler<Appliance, PlugwiseHAThingConfig> {
+
+    // private PlugwiseHAThingConfig config = new PlugwiseHAThingConfig();
+    private @Nullable Appliance appliance;
+    private final Logger logger = LoggerFactory.getLogger(PlugwiseHAApplianceHandler.class);
+
+    // Constructor
+
+    public PlugwiseHAApplianceHandler(Thing thing) {
+        super(thing);
+    }
+
+    public static boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return PlugwiseHABindingConstants.THING_TYPE_APPLIANCE_VALVE.equals(thingTypeUID)
+                || PlugwiseHABindingConstants.THING_TYPE_APPLIANCE_PUMP.equals(thingTypeUID)
+                || PlugwiseHABindingConstants.THING_TYPE_APPLIANCE_BOILER.equals(thingTypeUID)
+                || PlugwiseHABindingConstants.THING_TYPE_APPLIANCE_THERMOSTAT.equals(thingTypeUID);
+    }
+
+    // Overrides
+
+    @Override
+    protected synchronized void initialize(PlugwiseHAThingConfig config) {
+        if (thing.getStatus() == INITIALIZING) {
+            logger.debug("Initializing Plugwise Home Automation appliance handler with config = {}", config);
+            if (!config.isValid()) {
+                updateStatus(OFFLINE, CONFIGURATION_ERROR,
+                        "Invalid configuration for Plugwise Home Automation appliance handler.");
+                return;
+            }
+            try {
+                PlugwiseHABridgeHandler bridge = this.getPlugwiseHABridge();
+                if (bridge != null) {
+                    PlugwiseHAController controller = bridge.getController();
+                    if (controller != null) {
+                        this.appliance = getEntity(controller, true);
+                        if (this.appliance != null) {
+                            if (this.appliance.isBatteryOperated()) {
+                                addBatteryChannels();
+                            }
+                            setApplianceProperties();
+                            updateStatus(ONLINE);
+                        }
+                    }
+                }
+            } catch (PlugwiseHAException e) {
+                updateStatus(OFFLINE, COMMUNICATION_ERROR, STATUS_DESCRIPTION_COMMUNICATION_ERROR);
+            }
+        }
+    }
+
+    @Override
+    public void thingUpdated(Thing thing) {
```

Comment:
Overriding this method should normally not be needed. When a thing configuration is updated initialize is called and you can do everthing in there. So maybe this method can be removed.